### ISSUE TITLE
Consider adding fine tuning BERT for text classification and sequence tagging?

### ISSUE THREAD

**Issue by gccome** [flairNLP/flair#470]
Currently flair supports feature-based transfer learning from BERT. Any plan on adding fine-tuning based module in near future? That would be a great addition!

**Comment by mauryaland** [comment]
I allow myself to highlight the freshly available paper [Parameter-Efficient Transfer Learning for NLP](https://arxiv.org/abs/1902.00751). It seems relevant to this topic for further disccusion.

**Comment by gccome** [comment]
@mauryaland interesting paper! Thanks for sharing!

**Comment by tc-wolf** [comment]
You could override BertEmbeddings
```
class TrainableBERT(BertEmbeddings):
    pass
```

and set `self.static_embeddings = False` in the constructor, and override the `_add_embeddings_internal` method to do the same thing as the current method but without `model.eval()` and the `with torch.no_grad()` block.

AFAICT this should allow fine-tuning on the full input embeddings.  You'd need to be able to set this to eval mode (and not calculating gradients) when done training and doing predictions, however.

**Comment by alanakbik** [comment]
Fine-tuning has been added to master branch (#1492) and will be part of next Flair release.
